Ajout de FT.py et PROMS.csv

Ssp29-93 opened the pull request:
fct permettant l'ouverture de la fenêtre côté FT.
Elle récupère via un prompt du texte et affiche la liste des PG correspondant, liste des PG initialement récupérée dans PROMS.csv (elle n'est pas encore capable de supprimer les PG ayant votés). Elle n'est pas encore capable de communiquer avec PG.py, il faudra créer un txt d'échange pour certaines données et avancer sur le main.py également. Le code n'est pas encore entièrement mis sous forme de fonctions pour être appellé dans main.py.

Ssp29-93 committed:
Ajout de FT.py et PROMS.csv

fct permettant l'ouverture de la fenêtre côté FT.
Elle récupère via un prompt du texte et affiche la liste des PG correspondant, liste des PG initialement récupérée dans PROMS.csv (elle n'est pas encore capable de supprimer les PG ayant votés).
Elle n'est pas encore capable de communiquer avec PG.py, il faudra créer un txt d'échange pour certaines données et avancer sur le main.py également.
Le code n'est pas encore entièrement mis sous forme de fonctions pour être appellé dans main.py.

Ssp29-93 commented:
Simon arrête de faire zarma tu dors et accepte ma requête par pit's

OSiG43 commented on PROMS.csv:
```diff
@@ -0,0 +1,5 @@
+Grimaldi Quentin Sospi 29
```
Il faut vraiment mettre d'autre séparateur que des espaces car c'est un truc a tout casser si un jour un type met un espace dans son prénom (genre Jean François)

Ssp29-93 replied:
Comme je l'ai dit sur le groupe mon programme gère ce problème. Dans tout les cas j'ai update avec le csv de Diane

OSiG43 commented on FT.py:
```diff
@@ -0,0 +1,76 @@
+            lb.delete(0,len(ListePG))
+            for item in ListePG :
+                lb.insert('end',item[:-1])
+    pass
```
A enlever : pass sert uniquement en attendant d'écrire du contenu.
par ex pour une fonction dont j'ai le nom mais pas encore le contenu:
def unefonction():
       passs

Ssp29-93 replied:
fait

OSiG43 reviewed (changes requested):
J'ai noté 2 trucs.

Par contre normalement on devrais pas merge cett branche.
Le fonctionnement par branche l'idée c'est que du merge seulement quand la fonctionnalité que tu développes est terminé.

Du coup pour que vous puissiez tous travailler, vs devriez faire une branche "structure" ou vous mettez toutes la structure de base (avec des fonctions dont vs avez définit les entrées et les sorties par ex)  puis on merge cette branche et c'est à partir de ça que vous travaillez

peufgeul commented on FT.py:
```diff
@@ -0,0 +1,76 @@
+#### Fonction de recherche dans la liste déroulante
+###
+
+def recherche(j)  :
```
l'argument j n'est utilisé à aucun endroit dans la fonction + rajouter commentaire pour détailler le fonctionnement global de la fonction

Ssp29-93 replied:
L'argument J n'est pas utilisé mais un argument était obligatoire pour lancer la fonction.
J'ai commenté en plus

peufgeul commented on FT.py:
```diff
@@ -0,0 +1,76 @@
+                lb.insert('end',item[:-1])
+    pass
+
+def occurencePG(chaine,mot) : # Fonction d'occurence utile pour la recherche
```
rajouter un commentaire (docstring) pour détailler le fonctionnement de la fonction, ses arguments et ce qu'elle renvoie. Pour une fonction de ce niveau on retrouve facillement à quoi sert chaine et mot (lequel recherche dans lequel) mais pour des fonctions plus complex cela sera chiant

Ssp29-93 replied:
fonction supprimée

peufgeul commented on FT.py:
```diff
@@ -0,0 +1,76 @@
+        if len(saisie.get())>0 : # Si il y a qque chose de rentré
+            lb.delete(0,len(ListePG)) # On commence par tout effacer
+            for i in range (0,len(ListePG)) :
+                if occurencePG(ListePG[i],saisie.get())>0 : # Et on ajoute toutes les occurences
```
on peut utiliser la méthode str.find() puisque le nombre d'occurence n'est pas important, on veut juste savoir si il y a au moins une présence

Ssp29-93 replied:
implémentée, trop l'habitude après la PT

peufgeul reviewed (changes requested):
quelques détail de code qu'il vaut mieux commencer à appliquer maintenant pour avoir quelque chose de beaucoup plus périn quand le nombre de ligne va augmenter

peufgeul commented on FT.py:
```diff
@@ -0,0 +1,76 @@
+#### Création de la liste déroulande des PG
+###
+
+lb = Listbox(root, listvariable=StringVar()) # Liste des PG correspondants
```
peut-etre mettre un nom plus explicite comme lbListePg.
Permet de savoir plus facilement pour quoi cette ListBox sera utilisée, et permettre d'éviter les confusions si d'autre ListBox sont utilisés par la suite

Ssp29-93 replied:
Listeaffiche

Ssp29-93 committed:
Update PROMS.csv

Changé par le csv n de belef's de Diane, on passe aux ; du coup

Ssp29-93 reviewed (commented):
fixé

Ssp29-93 committed:
Update FT.py

Changements des anciens + amélioration de la fonction de recherche : les espaces du FT sont gérés (ex : Jules Martin et Martin Jules donnent le même résultat)

Ssp29-93 committed:
Update PROMS.csv

maj du csv avec : Les majuscules normalisées et aucun caractère spécial (même si ça marche avec c'est toujours plus pratique)
Il faudra préciser ça sur la tuy's destinée à la sdt

Ssp29-93 committed:
Delete PROMS.csv

inutile mtn

Ssp29-93 committed:
Update FT.py

quelques changements mineurs, transformation en fonction (retours pas encore codés), retrait des commandes de créations de fenêtres pour les insérées dans le main.py

Ssp29-93 committed:
Update FT.py

N de changements, maintenant on a bien des fonctions appellables dans le main, ça marche :)
